src/jrd/replication/Config.cpp

```diff
@@ -391,8 +389,10 @@ void Config::enumerate(ReplicaList& replicas)
 				}
 				else if (key == "source_guid")
 				{
-					if (!config->sourceGuid.fromString(value))
+					Guid guid;
```

Comment:
I didn't liked this too much ;)
I mean, the public constructor leaving the class with invalid data.

If I'd not going to add a throwable constructor, I would then use a `static std::optional<Guid> Guid::fromString(const char*)`.

Reply:
This is doable of course, but the same problem exists with `dbb_guid` which by design always exists but has a delayed initialization (when the header page is read). Generating it always (and replacing afterwards) looks like wasting GUIDs for nothing ;-) Using `AutoPtr` for lazy initialization means an unnecessary memory allocation. Making it `std::optional` means that the whole `Guid` class should be made optional, due to default constructor behaviour, so we get back to my original code (just with `std::optional` hidden inside `Guid`).

Reply:
> Making it `std::optional` means that the whole `Guid` class should be made optional, due to default constructor behaviour

`std::optional` does not require the class to have a default constructor.